src/manifest.c

```diff
@@ -968,11 +986,12 @@ bool changed_includes(struct manifest *old, struct manifest *new)
  *
  * Note: this function should be called after match_manifests().
  */
-int remove_old_deleted_files(struct manifest *m1, struct manifest *m2)
+int remove_deprecated_files(struct manifest *m1, struct manifest *m2, void *compfunc)
```

Comment:
Just declare it properly in the first place,then you don't have to cast it later.